packages/nc-gui/components/account/Profile.vue

```diff
@@ -0,0 +1,116 @@
+<script lang="ts" setup>
+const { currentUser } = storeToRefs(useUsers())
+
+const isErrored = ref(false)
+const isTitleUpdating = ref(false)
+const form = ref({
+  title: '',
+  email: '',
+})
+
+const { updateUserProfile } = useUsers()
+const formValidator = ref()
+
+const formRules = {
+  title: [
+    { required: true, message: 'Name required' },
+    { min: 2, message: 'Name must be at least 2 characters long' },
+    { max: 60, message: 'Name must be at most 60 characters long' },
+  ],
+}
+
+const onSubmit = async () => {
+  const valid = await formValidator.value.validate()
+
+  if (!valid) return
+
+  if (isTitleUpdating.value) return
+
+  isTitleUpdating.value = true
+  isErrored.value = false
+
+  try {
+    await updateUserProfile({ attrs: { display_name: form.value.title } })
+  } catch (e: any) {
+    console.error(e)
+  } finally {
+    isTitleUpdating.value = false
+  }
+}
+
+const email = computed(() => currentUser.value?.email)
+
+watch(
+  () => currentUser.value?.display_name,
+  () => {
+    if (!currentUser.value?.display_name) return
+
+    form.value.title = currentUser.value.display_name
+    form.value.email = currentUser.value.email
+  },
+  {
+    immediate: true,
+  },
+)
+
+watch(
+  () => form.value.title,
+  async () => {
+    try {
+      isErrored.value = !(await formValidator.value.validate())
+    } catch (e: any) {
+      isErrored.value = true
+    }
+  },
+)
```

Comment:
Suggestion: This watch can be replaced with `finishFailed` or `validate` event if possible.

https://antdv.com/components/form/#events